### PR TITLE
feat(webhooks): [nan 1064] webhook on sync error

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Add in logic to send webhooks on sync failure

## Issue ticket number and link
NAN-1064

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/webhooks/lib/sync.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/* eslint-disable @typescript-eslint/unbound-method */
```
Comment (khaliqgant) [inline packages/webhooks/lib/sync.unit.test.ts:1]: Copied over from shared

File: packages/webhooks/lib/sync.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc.js';
+import type { LogContext } from '@nangohq/logs';
+import type { NangoSyncWebhookBody } from './types.js';
+import { deliver, shouldSend } from './utils.js';
+import { WebhookType } from './enums.js';
+
+dayjs.extend(utc);
+
+export const sendSync = async ({
```
Comment (khaliqgant) [inline packages/webhooks/lib/sync.ts:11]: Logic essentially copied over from shared

File: packages/shared/lib/services/sync/run.service.ts [hunk 3]
```diff
@@ -103,6 +106,7 @@ export default class SyncRun {
     recordsService: RecordsServiceInterface;
     dryRunService?: NangoProps['dryRunService'];
     slackNotificationService?: SlackService;
+    sendSyncWebhook?: (params: SendSyncParams) => Promise<void>;
```
Comment (TBonnin) [inline packages/shared/lib/services/sync/run.service.ts:109]: I think I am gonna propose to Bastien a `refactor run.service` project because it has become a mammoth class that does everything. Its `run` function is 500+ lines I think

File: packages/shared/lib/services/notification/webhook.service.ts [hunk 4]
```diff
@@ -4,12 +4,10 @@ import dayjs from 'dayjs';
 import utc from 'dayjs/plugin/utc.js';
```
Comment (TBonnin) [inline packages/shared/lib/services/notification/webhook.service.ts:4]: what still live in this file? Is everything moved over to the webhooks package?

File: packages/types/lib/api.ts [hunk 5]
```diff
@@ -76,3 +76,8 @@ export interface Endpoint<T extends EndpointDefinition> {
      */
     Reply: ResDefaultErrors | (T['Error'] extends ApiError<any> ? T['Error'] | T['Success'] : T['Success']);
 }
+
+export interface ErrorPayload {
+    type: string;
+    description: string;
+}
```
Comment (TBonnin) [inline packages/types/lib/api.ts:83]: Is this an api related type?

File: packages/shared/lib/services/notification/webhook.service.ts (hunk 4 above)
Comment (khaliqgant) [reply packages/shared/lib/services/notification/webhook.service.ts:4]: webhook forwarding logic which I'll bring over in another PR.

File: packages/types/lib/api.ts (hunk 5 above)
Comment (khaliqgant) [reply packages/types/lib/api.ts:83]: internal API I would say

File: packages/shared/lib/services/sync/run.service.ts (hunk 3 above)
Comment (khaliqgant) [reply packages/shared/lib/services/sync/run.service.ts:109]: Yeah, it is a beast. Definitely needed.

File: packages/types/lib/api.ts (hunk 5 above)
Comment (bodinsamuel) [reply packages/types/lib/api.ts:83]: if we can keep this file for http api, maybe rename the file if clearer 👌🏻

File: packages/shared/lib/services/sync/run.service.ts [hunk 6]
```diff
@@ -420,11 +461,10 @@ export default class SyncRun {
             }
 
             const startTime = Date.now();
+            const syncStartDate = new Date();
```
Comment (bodinsamuel) [inline packages/shared/lib/services/sync/run.service.ts:464]: Could be moved at the very beginning of the method or could be a property of the class directly

**Review by bodinsamuel — APPROVED** [review]
🦾 Minor comments

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/services/sync/run.service.ts (hunk 6 above)
Comment (khaliqgant) [reply packages/shared/lib/services/sync/run.service.ts:464]: Will make a note to follow up on this later
